### PR TITLE
feat(ramp): add NetworkSwitcher view

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

This PR adds a new screen for the ramp flows that handles the case the current selected network is not supported

**Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/1024246/fa2a138d-dfc3-453f-ab0c-6cd0f86a821d

**QA**
Open the buy flow using a deeplink (metamask://buy-crypto) or universal link (https://metamask.app.link/buy-crypto) in an unsupported network to see the new screen. 

Here's a dummy url with the links https://62hv4l.csb.app/

**Issue**

fixes #???

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by cortisiko** [comment]
@wachunei A few test scenarios to consider:

- Adding a custom network via RPC URL that is supported on ramps. Add polygon or BSC
    Go to the onramps flow
    Does the custom network appear in that list? I take it that the network should not appear since we are only showing the list of popular networks we support?


- This one is a bit edgy: Add a popular network like optimism
   Go through the on ramps flow but do not submit any transaction
   return back to the wallet
   Then networks view and remove optimism from your network list
   Go through the on ramps flow again. Does the network switcher view appear again? I assume it should?

**Comment by wachunei** [comment]
> @wachunei A few test scenarios to consider:
> 
> * Adding a custom network via RPC URL that is supported on ramps. Add polygon or BSC
>   Go to the onramps flow
>   Does the custom network appear in that list? I take it that the network should not appear since we are only showing the list of popular networks we support?

You would not see the list because this view only shows up on unsupported networks. 

> * This one is a bit edgy: Add a popular network like optimism
>   Go through the on ramps flow but do not submit any transaction
>   return back to the wallet
>   Then networks view and remove optimism from your network list
At this point the wallet is back to a different network, let's call it `network_1`
>   Go through the on ramps flow again. Does the network switcher view appear again? I assume it should?
It should appear only if `network_1` is not supported.

Keep in mind supported or not supported is coming from an SDK method call that makes an API call so it is dynamic.

Here are videos for the two cases:

https://github.com/MetaMask/metamask-mobile/assets/1024246/9298f5e9-4bfc-49db-99a0-1c2bdd93fc00


https://github.com/MetaMask/metamask-mobile/assets/1024246/43cf2eda-6dec-4372-9a98-954dd5ab0d7e

File: app/components/UI/Ramp/common/hooks/useRampNetworksDetail.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+function useRampNetworksDetail() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error>();
+  const [networksDetails, setNetworksDetails] = useState<Network[]>([]);
+  const getNetworksDetail = useCallback(async () => {
+    try {
+      setError(undefined);
+      setIsLoading(true);
+      // TODO: change this mocked response to the real one once the SDK method is ready
+      // const networkDetails = await SDK.getNetworksDetails();
```
Comment (wachunei) [inline app/components/UI/Ramp/common/hooks/useRampNetworksDetail.ts:16]: This PR will remain as draft until on-ramp-sdk is upgraded to include this feature.

Comment (wachunei) [reply app/components/UI/Ramp/common/hooks/useRampNetworksDetail.ts]: Done in [ae29ffe](https://github.com/MetaMask/metamask-mobile/pull/7169/commits/ae29ffebdb1bf32590efc0f032be264bfd12e31b)

**Comment by wachunei** [comment]
@SocketSecurity ignore @consensys/on-ramp-sdk@1.24.0-beta.6 @types/is-url@1.2.30

Ignoring these warnings.

**Review by georgeweiler — APPROVED** [review]

**Review by NicolasMassart — COMMENTED** [review]
Looks good to me.

**Pull request merged**